src/main/java/dansplugins/factionsystem/services/ConfigService.java

```diff
@@ -163,7 +166,9 @@ public void handleVersionMismatch() {
         if (!getConfig().isInt("teleportDelay")) {
             getConfig().addDefault("teleportDelay", 3);
         }
-
+        if (!getConfig().isString("factionless")) {
+            getConfig().addDefault("factionless", "FactionLess");
```

Comment:
Perhaps `getConfig().set(...)` would be better here, or change the check to be `contains` rather than `isString`.
`addDefault` in combination with `options().copyDefaults()` will work if the key is not set, but that's not what we're checking - we're checking that the key _isn't a string_ - so in the case some other type is there, this will do nothing at all (as the default has already been overridden)

Reply:
i follow the methods above, maybe i should change the whole method in there?

Reply:
This is an old habit of mine. We can replace any instances of .addDefault() with .set(), that's what I do with more recent projects now.